### PR TITLE
Use Dune, update opam files

### PR DESCRIPTION
**Pull request by lindig**
- use Dune
- update opam files

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,27 +4,27 @@ OPAM_LIBDIR=$(DESTDIR)$(shell opam config var lib)
 .PHONY: build release install uninstall clean test doc reindent
 
 release:
-	    jbuilder build @install -j $$(getconf _NPROCESSORS_ONLN)
+	    dune build @install -j $$(getconf _NPROCESSORS_ONLN)
```
Comment (edwintorok) [inline Makefile]: Should we rename the `jbuild` file to `dune` as well while we're here?

File: xcp.opam [hunk 2]
```diff
@@ -5,34 +5,35 @@ bug-reports: "https://github.com/xapi-project/xcp-idl/issues"
...
   "message-switch-core"
   "message-switch-unix"
-  "ocamlfind" {build}
-  "ounit" {>= "2.0.0"}
+  "ocaml-migrate-parsetree"
+  "ounit"
+  "ppx_deriving_rpc"
   "ppx_sexp_conv"
   "re"
-  "rpc" {>= "5.0.0"}
```
Comment (edwintorok) [inline xcp.opam]: Are you sure it still works without this?

Comment (lindig) [reply xcp.opam]: Travis says yes. At least in our environment, we should never have incompatible versions.

File: Makefile (hunk 1 above)
Comment (lindig) [reply Makefile]: We are using the `Jbuild_plugin` - is that available in Dune, too?

Comment (edwintorok) [reply Makefile]: Yes, but with some caveats. We can't use the Unix module, so Unix.getenv has to be replaced by Sys.getenv.
We have to drop the external calls to ocamlfind and the flags variable (its not needed anyway with the new pps calls).
And then something changed around how many parenthesis you need in dune vs jbuild.
Here is an updated network/dune that should work:
```
(* -*- tuareg -*- *)
let rewriters_ppx = ["ppx_deriving_rpc"]

let coverage_rewriter =
  let is_coverage = try Sys.getenv "BISECT_ENABLE" = "YES" with Not_found -> false in
  if is_coverage then
    "(preprocess (pps (bisect_ppx -conditional)))"
  else
    Printf.sprintf "(preprocess (pps %s))" (String.concat " " rewriters_ppx)

let () = Printf.ksprintf Jbuild_plugin.V1.send {|
(library
 (name xcp_network)
  (public_name xcp.network)
  (flags (:standard -w -39-33))
  (modules (:standard \ network_cli))
  (libraries
   rpclib
    threads
    xcp)
  (wrapped false)
  %s)

(executable
 (name network_cli)
  (modules network_cli)
  (libraries
    cmdliner
    rpclib.cmdliner
    rpclib.markdown
    xcp.network))

(alias
 (name runtest)
   (deps network_cli.exe)
     (action (run ${<})))

|} coverage_rewriter
```

Comment (lindig) [reply Makefile]: This is not just renaming files - Dune demands all kinds of changes that I would rather not do now.

Comment (edwintorok) [reply Makefile]: Ok, lets do them in a future PR.

**Pull request merged**